showcase/app/controllers/page-components/form/masked-input.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Controller from '@ember/controller';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { deepTracked } from 'ember-deep-tracked';
+
+import type { PageComponentsFormMaskedInputModel } from 'showcase/routes/page-components/form/masked-input';
+
+export default class PageComponentsFormMaskedInputController extends Controller {
+  declare model: PageComponentsFormMaskedInputModel;
+
+  @deepTracked fieldValues = {
```

Comment:
I wonder why we use ember-deep-tracked over something like TrackedObject from tracked-builtins ?

Reply:
Personally, when I tried TrackedObject, it didn't work for the data structure that I had (an object with nested arrays)

Reply:
I will circle back to all instance of this after ts migration in order not to cause conflict but for visibility I have nothing against using this pattern but int should be considered antipattern and should be used very selectively, also convo in slack https://hashicorp.slack.com/archives/C7KTUHNUS/p1753416952505689 , also if you ever going to work on any production code, definitely do not use this